src/Raven.Server/Rachis/FollowerAmbassador.cs

```diff
@@ -651,11 +659,49 @@ private unsafe void WriteSnapshotToFile(ClusterOperationContext context, Stream
                                                 var currentTreeValueKey = treeIterator.CurrentKey;
                                                 binaryWriter.Write(currentTreeValueKey.Size);
                                                 copier.Copy(currentTreeValueKey.Content.Ptr, currentTreeValueKey.Size);
-                                                var reader = treeIterator.CreateReaderForCurrent();
-                                                binaryWriter.Write(reader.Length);
-                                                copier.Copy(reader.Base, reader.Length);
-
-                                                totalSizeInBytes += sizeof(long) + currentTreeValueKey.Size + reader.Length;
+                                                totalSizeInBytes += sizeof(int) + currentTreeValueKey.Size;
+
+                                                switch (type)
+                                                {
+                                                    case TreeFlags.None:
+                                                    case TreeFlags.FixedSizeTrees:
```

Comment:
When we have `TreeFlags.FixedSizeTrees` shouldn't we open it as `FixedSizeTree` and then iterate over its values? Note that `FixedSizeTree` can have two types:

- RootObjectType.EmbeddedFixedSizeTree
- RootObjectType.FixedSizeTree

My concern is that it works currently only with embedded type

Reply:
Shouldn't this entire code be more like in compaction since we need to deal with different types of trees:

https://github.com/ravendb/ravendb/blob/ef53f2f4754c4458021bef21764bc0479ea725ea/src/Voron/Impl/Compaction/StorageCompaction.cs#L157-L179